### PR TITLE
Fixing some issues with depth tensor parallelism

### PR DESCRIPTION
**Pull request by siddharth9820**
1. Removed the all-reduce on bias. This was causing performance issues.
2. Added a function to all-reduce gradients of bias and other non tensor parallel parameters which should fix the convergence bug
3. Creating a combined row+tensor parallel group. This is used by megatron-axonn to broadcast data correctly.

### CODE REVIEW

**Comment by siddharth9820** [comment]
@prajwal1210 can we test with this branch again? Also welcoming suggestions for formatting the __init__.py file.

File: axonn/axonn.py [hunk 1]
```diff
@@ -137,6 +137,11 @@ def init(
     config.inter_layer_parallel_rank = comm_handle.inter_layer_parallel_rank
     config.data_parallel_rank = comm_handle.data_parallel_rank
     config.intra_layer_parallel_rank = comm_handle.intra_layer_parallel_rank
+    config.intra_layer_depth_parallel_rank = comm_handle.intra_layer_depth_parallel_rank
+    config.intra_layer_row_parallel_rank = comm_handle.intra_layer_row_parallel_rank
+    config.intra_layer_column_parallel_rank = (
```
Comment (prajwal1210) [inline axonn/axonn.py:142]: Just curious on why the different formatting?

Comment (siddharth9820) [reply axonn/axonn.py:142]: automated by black

**Comment by siddharth9820** [comment]
Moving to another branch

**Pull request closed without merging**